misc/services/mobile-webui/mobile-webui-frontend/src/components/UpdateCheck.jsx

```diff
@@ -49,9 +50,11 @@ class UpdateCheck extends Component {
   };
 
   componentDidMount() {
-    const { updateInterval } = this.props;
-    let intervalId = setInterval(this.checkServerVersion, updateInterval);
-    this.setState({ intervalId: intervalId });
+    if (REGISTER_SERVICE_WORKER) {
+      const { updateInterval } = this.props;
+      let intervalId = setInterval(this.checkServerVersion, updateInterval);
+      this.setState({ intervalId: intervalId });
+    }
```

Comment:
why not here?
![image](https://user-images.githubusercontent.com/1244701/140471175-375d057f-6950-4593-8bfe-5b6e3c823488.png)

Reply:
It will still mount the component and it will execute the code from componentDidMount wihin the UpdateCheck component.
So something like 

{!REGISTER_SERVICE_WORKER && <UpdateCheck updateInterval={UPDATE_CHECK_INTERVAL} />} won't work.

Also this check setting is specific to UpdateCheck component and I find the correct place to be is within it

Reply:
> {!REGISTER_SERVICE_WORKER && <UpdateCheck updateInterval={UPDATE_CHECK_INTERVAL} />} won't work.

why it won't work?

> Also this check setting is specific to UpdateCheck component and I find the correct place to be is within it

going with the same thinking, why we have `updateInterval` property. Why the UPDATE_CHECK_INTERVAL is not checked directly inside UpdateCheck component?

not suggesting to do so, but i need to understand, from design point of view, why UPDATE_CHECK_INTERVAL is checked outside of `UpdateCheck` component and `REGISTER_SERVICE_WORKER` is checked inside the component?